lasagne/layers/sparse/noise.py

```diff
@@ -0,0 +1,40 @@
+import theano
+import theano.sparse as sparse
+
+from ..noise import DropoutLayer as Dropout
+
+
+__all__ = [
+    "DropoutLayer",
+    "dropout",
+]
+
+
+class DropoutLayer(Dropout):
+    def get_output_for(self, input, deterministic=False, **kwargs):
+        """
+        Parameters
+        ----------
+        input : theano sparse matrix
+            output from the previous layer
+        deterministic : bool
+            If true dropout and scaling is disabled, see notes
+        """
+        if deterministic or self.p == 0:
+            return input
+        else:
+            assert type(input) == sparse.basic.SparseVariable
+
+            retain_prob = 1 - self.p
+            if self.rescale:
+                input = sparse.basic.mul(input, 1/retain_prob)
```

Comment:
In both occasions, instead of using `1`, you should use `lasagne.utils.floatX(1)` -- otherwise the result will be upcast to float64 in case `self.p` is a numpy matrix.